pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/DefaultGroupKeyGenerator.java

```diff
@@ -88,9 +97,16 @@
   private int _numGroupKeys = 0;
   // For LONG_MAP_BASED storage type.
   private Long2IntOpenHashMap _groupKeyToId;
+
+  // Reverse mapping for trimming group keys
+  private Int2LongOpenHashMap _idToGroupKey;
+
   // For ARRAY_MAP_BASED storage type.
   private Object2IntOpenHashMap<IntArrayList> _arrayGroupKeyToId;
 
+  // Reverse mapping for trimming group keys
+  private Int2ObjectOpenHashMap<IntArrayList> _idToArrayGroupKey;
```

Comment:
please group the data structures that get used after we reach the max capacity and add a comment